query/src/main/java/io/keen/client/java/FunnelStep.java

```diff
@@ -0,0 +1,225 @@
+        
+    // Required parameters
+    /**
+     * The name of the event collection.
+     */
+    private final String collectionName;
+    
+    /**
+     * The name of the actor property for this funnel step.
+     */
+    private final String actorPropertyName;
+    
+    /**
+     * The timeframe used to qualify the events used for this funnel step.
+     * If not specified here for the funnel step, must be specified on the funnel
+     * itself.
+     */
+    private final Timeframe timeframe;
+    
+    // Optional parameters
+    /**
+     * An optional list of filters used to qualify the events used in 
+     * this funnel step.
+     */
+    private final RequestParameterCollection<Filter> filters;
+    
+    // Special parameters
+    /**
+     * If true, inverted means events matching this step will be excluded from
+     * the funnel results.
+     * Cannot be set if this is the first step in a funnel query.
+     */
+    private final Boolean inverted;
+    
+    /**
+     * If optional is set, the funnel will ignore the effects of this step
+     * on subsequent steps.
+     */
+    private final Boolean optional;
+    
+    /**
+     * If set, the query result will include a list of actor property values
+     * for this step.
+     */
+    private final Boolean withActors;
+    
+    /**
+     * FunnelStep constructor including all required, optional, and special parameters.
+     * Any parameter that is optional may be set to null.
+     *
+     * @param collectionName    The name of the event collection for the funnel step.
+     * @param actorPropertyName The name of the actor property for the funnel step.
+     * @param timeframe         The timeframe for the set of events to include,
+     *                          optional if a timeframe is set for the entire funnel.
+     * @param filters           Optionally, any filters to use for the set of events
+     *                          to include in this step.
+     * @param inverted          When set to true will exclude events matching
+     *                          this step from the funnel. Optional and cannot
+     *                          be set to true for the first funnel step
+     * @param optional          When set to true the funnel will not apply the
+     *                          effects of this step to subsequent steps.
+     *                          Optional and cannot be set to true for the first
+     *                          funnel step. 
+     * @param withActors        Optional, and when set to true will include values
+     *                          for the actor property for this step in the results
+     *                          of the query.
+     */
+    public FunnelStep(
+        final String collectionName,
+        final String actorPropertyName,
+        final Timeframe timeframe,
+        final Collection<Filter> filters,
+        final Boolean inverted,
+        final Boolean optional,
+        final Boolean withActors) {
```

Comment:
I think I'd like to try to clean this up to make it more readable from calling code. Either something with flags or Enums for special parameters I think would help.